build/utils/npm.go

```diff
@@ -9,189 +9,284 @@ import (
+			log.Error("couldn't calculate checksum for : '" + dep.Id + "'. Try to delete 'node_models' and/or 'package-lock.json'.")
+			return nil, err
 		}
-		wg.Done()
-	}()
 
-	producerConsumer.Run()
-	if err != nil {
-		return
+		dependenciesList = append(dependenciesList, dep.Dependency)
 	}
-	close(dependenciesChan)
-	wg.Wait()
-
-	// Read the error from the channel, but don't wait if the channel buffer is empty.
-	select {
-	case err = <-errorChan:
-	default:
-		return
+	if len(missingPeerDeps) > 0 {
+		log.Info("The following dependencies will not be included in the build-info, because the 'npm ls' command did not return its integrity.\nThe reason why the version wasn't returned may be because the package is a 'peerDependency', which was not manually installed.\n It is therefore okay to skip this dependency: \n" + strings.Join(missingPeerDeps, "\n"))
+	}
+	if len(missingBundledDeps) > 0 {
+		log.Info("The following dependencies will not be included in the build-info, because the 'npm ls' command did not return its integrity.\nThe reason why the version wasn't returned may be because the package is a 'bundleDependencies', which was not manually installed.\n It is therefore okay to skip this dependency." + strings.Join(missingBundledDeps, "\n"))
 	}
 	return
 }
 
-// createHandlerFunc creates a function that runs traverseDependenciesFunc (if it's not nil) with dep as its parameter.
-// If traverseDependenciesFunc returns false, then dep will not be saved in the module's dependencies list.
-func createHandlerFunc(dep *entities.Dependency, dependenciesChan chan *entities.Dependency, traverseDependenciesFunc func(dependency *entities.Dependency) (bool, error)) func(threadId int) error {
-	return func(threadId int) error {
-		var err error
-		saveDep := true
-		if traverseDependenciesFunc != nil {
-			saveDep, err = traverseDependenciesFunc(dep)
-			if err != nil {
-				return err
-			}
-		}
-		if saveDep {
-			dependenciesChan <- dep
-		}
-		return nil
-	}
+type prototypeNode struct {
+	entities.Dependency
+	*npmLsDependency
 }
 
-// Run npm list and parse the returned JSON.
-// typeRestriction must be one of: 'dev' or 'prod'!
-func prepareDependencies(typeRestriction, executablePath, srcPath, moduleId string, npmArgs []string, results *map[string]*entities.Dependency, log utils.Log) error {
-	// Run npm list
-	// Although this command can get --development as a flag (according to npm docs), it's not working on npm 6.
-	// Although this command can get --only=development as a flag (according to npm docs), it's not working on npm 7.
-	data, errData, err := runList(typeRestriction, executablePath, srcPath, npmArgs, log)
+// Run 'npm list ...' command and parse the returned result to create a dependencies map of.
+// The dependencies map looks like name:version -> entities.Dependency
+func CalculatePrototypeTree(executablePath, srcPath, moduleId string, npmArgs []string, cacache *cacache, log utils.Log) (map[string]*prototypeNode, error) {
```

Comment:
Looks like it is not in use
```suggestion
func CalculatePrototypeTree(executablePath, srcPath, moduleId string, npmArgs []string, log utils.Log) (map[string]*prototypeNode, error) {
```